### PR TITLE
Updated to node v7.8.0

### PR DESCRIPTION
**Pull request by mcollina**
As titled, this is based on the latest build scripts from readable-stream.

It needs an iteration on the README.

On which version do we want to publish? 2 as `readable-stream`?
I think it deserves a major bump.

cc @calvinmetcalf

### CODE REVIEW

**Comment by mcollina** [comment]
@rvagg can you please enable travis, so that we can test in all platforms?

**Comment by calvinmetcalf** [comment]
it would be 1.0.0 if we were going by readable stream

**Comment by mcollina** [comment]
This is the current result from `npm show string_decoder`:

```js
{ name: 'string_decoder',
  description: 'The string_decoder module from Node core',
  'dist-tags': { latest: '0.10.31' },
  versions:
   [ '0.0.0',
     '0.0.1',
     '0.10.24',
     '0.10.25-1',
     '0.10.25',
     '0.10.31',
     '0.11.10-1',
     '0.11.10' ],
  maintainers: [ 'substack <mail@substack.net>', 'rvagg <rod@vagg.org>' ],
  time:
   { modified: '2016-11-01T18:40:08.313Z',
     created: '2013-12-03T20:13:26.191Z',
     '0.0.0': '2013-12-03T20:13:27.743Z',
     '0.0.1': '2014-01-17T19:53:23.612Z',
     '0.10.24': '2014-01-18T05:27:35.195Z',
     '0.11.10': '2014-01-18T05:28:38.924Z',
     '0.10.25': '2014-01-28T02:21:38.263Z',
     '0.11.10-1': '2014-01-28T02:24:31.052Z',
     '0.10.25-1': '2014-02-12T09:51:56.014Z',
     '0.10.31': '2014-08-23T04:25:19.329Z' },
  repository:
   { type: 'git',
     url: 'git://github.com/rvagg/string_decoder.git' },
  readmeFilename: 'README.md',
  users:
   { wenbing: true,
     simplyianm: true,
     xgqfrms: true,
     mojaray2k: true },
  homepage: 'https://github.com/rvagg/string_decoder',
  keywords: [ 'string', 'decoder', 'browser', 'browserify' ],
  bugs: { url: 'https://github.com/rvagg/string_decoder/issues' },
  license: 'MIT',
  version: '0.10.31',
  main: 'index.js',
  dependencies: {},
  devDependencies: { tap: '~0.4.8' },
  scripts: { test: 'tap test/simple/*.js' },
  gitHead: 'd46d4fd87cf1d06e031c23f1ba170ca7d4ade9a0',
  dist:
   { shasum: '62e203bc41766c6c28c9fc84301dab1c5310fa94',
     tarball: 'https://registry.npmjs.org/string_decoder/-/string_decoder-0.10.31.tgz' },
  directories: {} }
```

**Comment by mcollina** [comment]
@calvinmetcalf can you review this? Are you ok to release as v1.0.0?

File: build/files.js [hunk 1]
```diff
@@ -8,29 +8,78 @@
...
         + '\n  || function(encoding) {'
+        + '\n       encoding = \'\' + encoding'
         + '\n       switch (encoding && encoding.toLowerCase()) {'
         + '\n         case \'hex\': case \'utf8\': case \'utf-8\': case \'ascii\': case \'binary\': case \'base64\': case \'ucs2\': case \'ucs-2\': case \'utf16le\': case \'utf-16le\': case \'raw\': return true;'
         + '\n         default: return false;'
         + '\n       }'
         + '\n     }'
         + '\n'
 
+        + '\nfunction _normalizeEncoding(enc) {'
```
Comment (calvinmetcalf) [inline build/files.js:43]: if your doing it from scratch I highly recomend tempalte strings instead of this stuff, one million times easier to edit

**Review by calvinmetcalf — APPROVED** [review]
not using template strings isn't really a blocker, we can move over to them in the future pretty easily

File: build/files.js (hunk 1 above)
Comment (mcollina) [reply build/files.js:43]: I agree, but I copied and pasted this :).

**Pull request merged**